addons/website_event/static/src/interactions/website_event_slot_details.js

```diff
@@ -0,0 +1,141 @@
+        const nextBtn = this.el.querySelector(".o_wevent_slot_next button");
+        if (dates.length <= this.datesLimitPerPage) {
+            [previousBtn, nextBtn].forEach((button) => button.classList.add("d-none"));
+        } else {
+            [previousBtn, nextBtn].forEach((button) => {
+                button.classList.remove("disabled");
+            });
+            if (this.currentSlotPage === 0) {
+                previousBtn.classList.add("disabled");
+            } else if(max >= dates.length) {
+                nextBtn.classList.add("disabled");
+            }
+        }
+    }
+
+    /**
+     * Update the displayed slots page.
+     * @param {MouseEvent} ev
+     */
+    _onChangePageClick(ev) {
+        const numPage = ev.currentTarget.classList.contains("o_wevent_slot_next") ? this.currentSlotPage + 1 : this.currentSlotPage - 1;
+        const maxNumPage = Math.ceil(this.el.querySelectorAll(".o_wevent_slot_date").length / this.datesLimitPerPage) - 1;
+        if (numPage < 0 || numPage > maxNumPage) {
+            return;
+        }
+        this._changePage(numPage);
+    }
+
+    /**
+     * Reset slot selection
+     * @param {MouseEvent} ev
+     */
+    _onClose(ev) {
+        this.el.querySelectorAll(".o_wevent_slot_btn").forEach((btn) => btn.classList.replace("btn-primary", "btn-light"));
+        this.selectedSlotTitle.textContent = _t("Select a Date:");
+        this.selectedSlotDatetime.textContent = "";
+        this.form.removeAttribute("data-selected-slot-id");
+    }
+
+    /**
+     * Select a slot
+     * @param {MouseEvent} ev
+     */
+    _onSlotSelected(ev) {
+        // Visually select the button
+        this.el.querySelectorAll(".o_wevent_slot_btn").forEach((btn) => btn.classList.replace("btn-primary", "btn-light"));
+        ev.currentTarget.classList.replace("btn-light", "btn-primary");
+        // Change selected slot title
+        const selectedDatetime = DateTime.fromISO(ev.currentTarget.dataset.slotStart.replace(" ", "T"), { zone: "UTC" });
+        const localizedDatetime = selectedDatetime.setZone(ev.currentTarget.dataset.eventTz);
+        this.selectedSlotTitle.textContent = _t("Selected Date: ");
+        this.selectedSlotDatetime.textContent = localizedDatetime.toFormat("MMM dd yyyy, EEEE, h:mm a");
+        this.form.setAttribute("data-selected-slot-id", parseInt(ev.currentTarget.dataset.slotId));
+    }
+
+    /**
+     * @param {MouseEvent} ev
+     */
+    async _onSubmitClick(ev) {
+        const formEl = ev.currentTarget.closest("form");
+        this.buttonDisabled = true;
+        const modal = await this.waitFor(rpc(
+            formEl.action,
+            Object.assign(
+                {},
+                Object.fromEntries(new FormData(formEl)),
+                {'event_slot_id': this.selectedSlot}
+            )
+        ));
+        const modalEl = new DOMParser().parseFromString(modal, "text/html").body.firstChild;
```

Comment:
Was going to say that this looks very cryptic but I see it's already done like that in the other one already so... let's keep it :smile: